rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala

```diff
@@ -2059,4 +2056,284 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
     result.exprs should be(Seq(Expr(resultList)))
     errorLog.readAndClearErrorVector().unsafeRunSync should be(Vector.empty[InterpreterError])
   }
+  "toSet" should "transform [1, 2, 3] into Set(1, 2, 3)" in {
```

Comment:
Please add a test that verifies duplicated elements are removed from the source collection when the source collection is transformed into a set.